distributed/scheduler.py

```diff
@@ -2329,7 +2329,7 @@ def transition_processing_erred(
 
             w = _remove_from_processing(self, ts)
 
-            ts.erred_on.add(w or worker)  # type: ignore
+            ts.erred_on.add(w)
```

Comment:
IIUC this is what motivated your change. However, if `_remove_from_processing` returns None, we'd use `worker` instead. Is `worker` also None in this scenario? If so, is the kwarg even still in use?

Reply:
Since we encountered a `None` in `ts.erred_on` in #6874, there needs to be _some_ scenario where both `w` and `worker`_ are `None`. I dropped the `worker` kwarg since it's not being used anywhere and will get collected in `**kwargs`.

Reply:
I'm a bit worried here about consistency. 

The only way I am aware of to trigger this transition is by receiving a `task-erred` message from a worker https://github.com/dask/distributed/blob/1d0701b51c1dc354fb22af9b27b8f722f02daf63/distributed/scheduler.py#L3093

which is handled by handle_worker which will always have a `worker` extra kwarg 

https://github.com/dask/distributed/blob/1d0701b51c1dc354fb22af9b27b8f722f02daf63/distributed/scheduler.py#L4888

https://github.com/dask/distributed/blob/1d0701b51c1dc354fb22af9b27b8f722f02daf63/distributed/core.py#L842

i.e. calling `handle_task_erred` will always have `worker` as part of `msg` (proper signatures + mypy would help here)

https://github.com/dask/distributed/blob/1d0701b51c1dc354fb22af9b27b8f722f02daf63/distributed/scheduler.py#L4756-L4757

which leads us to the actual stimulus handler which I think should verify if the task-erred message actually originated from the worker the task is currently processing on

Reply:
Ok, the `worker=None` is coming from https://github.com/dask/distributed/blob/1d0701b51c1dc354fb22af9b27b8f722f02daf63/distributed/scheduler.py#L4308-L4315

therefore, the scenario of an empty `ts.erred_on` comes from tasks that have been transitioned to error because of a `KilledWorker` exception.

Reply:
> therefore, the scenario of an empty ts.erred_on comes from tasks that have been transitioned to error because of a KilledWorker exception.

That makes sense, #6874 includes a bunch of `KilledWorker` exceptions.